### PR TITLE
Add ArcGis ImageServer support

### LINKED ISSUE #1458: Support ArcGIS ImageServer
Eg http://data.actmapi.act.gov.au/arcgis/rest/services/actmapi/imagery2015mga/ImageServer

(Although I can't get that one to actually show anything in Esri's online viewer.)

### PR DESCRIPTION
**Pull request by nf-s**
### Add ArcGis ImageServer support

Fixes https://github.com/TerriaJS/terriajs/issues/1458

Todo

- [x] Support tiled endpoints (similar to MapServer)
- [x] Look into multi-dim
- [x] Look into raster functions
- [x] Look into color maps
- [x] Get feature picking working
- [x] Test time-enabled ImageServer
- [x] Remove selectable dimensions and multi-dim/raster function/color map fns
- [x] Test token-based auth
- [x] ~Integrate with ArcGis REST...~
- [x] Write unit tests

### Test me

- [Test link](http://ci.terria.io/image-server/#clean&start=%7B%22initSources%22%3A%20%5B%7B%22homeCamera%22%3A%20%7B%22north%22%3A%20-8%2C%22east%22%3A%20158%2C%22south%22%3A%20-45%2C%22west%22%3A%20109%7D%2C%22catalog%22%3A%20%5B%7B%22name%22%3A%20%22QLD%20basemap%22%2C%22url%22%3A%20%22https%3A%2F%2Fspatial-img.information.qld.gov.au%2Farcgis%2Frest%2Fservices%2FBasemaps%2FLatestSatelliteWOS_AllUsers%2FImageServer%22%2C%22type%22%3A%20%22esri-imageServer%22%7D%2C%7B%22name%22%3A%20%22Time%20enabled%20-%20with%20multi-dimensions%22%2C%22url%22%3A%20%22https%3A%2F%2Fimage.wsapi.cloud.bom.gov.au%2Farcgis%2Frest%2Fservices%2Fawra_historical%2Fphysical_day_crf_multivariable_archive_1985_2004_data%2FImageServer%22%2C%22type%22%3A%20%22esri-imageServer%22%2C%22renderingRule%22%3A%20%7B%22rasterFunction%22%3A%20%22e0_day%22%7D%7D%2C%7B%22name%22%3A%20%22Tiled%20example%20(uses%20%60tile%60%20endpoint)%22%2C%22url%22%3A%20%22https%3A%2F%2Fimagery.oregonexplorer.info%2Farcgis%2Frest%2Fservices%2FOSIP_2022%2FOSIP_2022_WM%2FImageServer%22%2C%22type%22%3A%20%22esri-imageServer%22%7D%5D%2C%22viewerMode%22%3A%20%223dSmooth%22%2C%22baseMaps%22%3A%20%7B%22defaultBaseMapId%22%3A%20%22basemap-positron%22%2C%22previewBaseMapId%22%3A%20%22basemap-positron%22%7D%7D%5D%7D)

Extra links to test with "Add Web Data"

- https://sampleserver6.arcgisonline.com/arcgis/rest/services/CharlotteLAS/ImageServer
- https://vicmap.land.vic.gov.au/agsimage/rest/services/elevation/vicmap_dem10m_v5m_ahd_epsg7844/ImageServer
- https://www.lmbc.nsw.gov.au/arcgis/rest/services/NVR/SensitiveRegulatedLand/ImageServer
- https://arcgis.asdc.larc.nasa.gov/server/rest/services/POWER/power_901_annual_meterology_utc/ImageServer

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [x] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+      new Date(
+        imageServerStratum.imageServer.timeInfo.timeExtent[1]
+      ).toISOString(),
+      undefined,
+      this.maxRefreshIntervals
+    );
+    return result;
+  }
+
+  private get _currentImageryParts(): ImageryParts | undefined {
```
Comment (na9da) [inline lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts:423]: Nitpick: might be better to also make this `@computed`

File: lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts [hunk 2]
```diff
@@ -0,0 +1,614 @@
...
+function getBaseURI(item: ArcGisImageServerCatalogItem) {
+  const uri = new URI(item.url);
+  const lastSegment = uri.segment(-1);
+  if (lastSegment && lastSegment.match(/\d+/)) {
+    uri.segment(-1, "");
+  }
+  return uri;
+}
+
+async function getJson(item: ArcGisImageServerCatalogItem, uri: any) {
```
Comment (na9da) [inline lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts]: Nitpick: Instead of `any` we could use `uri.URI` type.

**Review by na9da — CHANGES_REQUESTED** [review]
@nf-s - the implementation looks very thorough. 
I have left a couple of suggestions.

File: lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts (hunk 1 above)
Comment (nf-s) [reply lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts:423]: Done!

File: lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts (hunk 2 above)
Comment (nf-s) [reply lib/Models/Catalog/Esri/ArcGisImageServerCatalogItem.ts]: Done!

**Review by na9da — APPROVED** [review]
Looks good!

**Pull request merged**